src/Common/EventStore/FollowEventStoreDispatcher.php

```diff
@@ -15,13 +15,12 @@ final class FollowEventStoreDispatcher
 
     public function __construct(
         private readonly PollableEventStore $pollableEventStore,
-        private EventStorePointer $eventStorePointer,
+        private readonly EventStorePointer $eventStorePointer,
         private readonly StoredEventSubscriber $storedEventSubscriber,
         private readonly int $batchSize,
         private readonly int $throttleTimeInMilliseconds,
         private readonly EventDispatcherInterface $eventDispatcher
     ) {
-        $this->eventStorePointer = new InMemoryCacheEventStorePointer($eventStorePointer);
```

Comment:
Implementations need to cache on their own now, because the usage of `NullEventStorePointerFactory` with disabled `GapDetection` and immediate deleted events could lead to missing events. E.g., when events 1, 2, 4, 5 are fetched and 3 is still in an uncommitted transaction, the cache decorator would store 5. In the meantime the transaction with event 3 is committed. In sub-sequent runs, it would use that cached value to read the next batch of events (`where id > 5`). Event 3 will never be consumed. Not happy about this, but all "real" `EventStorePointer`s need to cache on their own now.

As this causes friction, maybe `GapDetection` should never be disabled.

Reply:
Change is reverted for now. Will open another task in #79.